packages/botonic-nlp/src/preprocess/preprocessor.ts

```diff
@@ -0,0 +1,72 @@
+import { Locale } from '../types'
+import { PaddingPosition, PreprocessEngines } from './types'
+
+export class Preprocessor {
+  engines: PreprocessEngines = {}
+
+  constructor(
+    public readonly locale: Locale,
+    public readonly maxLength: number,
+    public readonly paddingPosition: PaddingPosition = 'post'
+  ) {
+    this.loadEngine('normalizer')
+    this.loadEngine('tokenizer')
+    this.loadEngine('stopwords')
+    //   this.loadEngine('stemmer')
```

Comment:
add a TODO to handle this scenario in the future depending if embeddings are present or not